src/redux/reducers/compare.ts

```diff
@@ -359,27 +407,52 @@ export const selectComparisonListItems = createSelector(
   }
 );
 
+/* * * * * * * * * * * * * *
+ * Utilities
+ * * * * * * * * * * * * * */
+export function isCompleteResourceSet(options: PartialResourceSet | undefined): options is ResourceSet {
+  switch (options?.type) {
+    case 'local':
+      return true;
+    case 'cluster':
+      return isDefined(options.context);
+    case 'kustomize':
+      return isDefined(options.kustomizationId);
+    case 'helm':
+      return isDefined(options.chartId) && isDefined(options.valuesId);
+    default:
+      return false;
+  }
+}
+
 /* * * * * * * * * * * * * *
  * Listeners
  * * * * * * * * * * * * * */
 export const resourceFetchListener =
   (side: CompareSide): AppListenerFn =>
   listen => {
     listen({
-      predicate: action => {
+      predicate: (action, state) => {
         const resourceSetUpdated = isAnyOf(resourceSetSelected, resourceSetRefreshed, resourceSetCleared);
         if (!resourceSetUpdated(action)) return false;
+
         const actionSide = action.payload.side;
-        return actionSide === side;
+        if (actionSide !== side) return false;
+
+        const resourceSet = selectResourceSet(state.compare, side);
+        if (!isCompleteResourceSet(resourceSet)) return false;
```

Comment:
So this is the whole essence of allowing this kind of user experience.

The configuration in our store is kept partially and only once it's completely present do we move on to the next stage.